File: frontend/src/pages/PersonalUserPage.js
import React, { useState, useEffect } from "react";
import axios from "axios";
import {
  Box,
  Heading,
  FormControl,
  FormLabel,
  Input,
  Button,
  VStack,
  useToast,
  AlertDialog,
  AlertDialogBody,
  AlertDialogFooter,
  AlertDialogHeader,
  AlertDialogContent,
  AlertDialogOverlay,
  SimpleGrid,
  Divider, // <-- new
} from "@chakra-ui/react";

const PersonalUserPage = () => {
  const toast = useToast();
  const cancelRef = React.useRef();

  const [formData, setFormData] = useState({
    profile_picture: null,
    emergency_contact_name: "",
    emergency_contact_number: "",
    full_name: "",
    email: "",
    phone_number: "",
    date_of_birth: "",
    national_id_number: "",
    gender: "",
    address_street: "",
    address_city: "",
    address_province: "",
    address_postal_code: "",
    employment_status: "",
    occupation: "",
    monthly_income: "",
    employer_name: "",
    bank_name: "",
    bank_account_number: "",
    bank_account_type: "",
  });

  const [isOpen, setIsOpen] = useState(false);

  useEffect(() => {
    const token = localStorage.getItem("token");
    axios
      .get("http://localhost:5000/profile/me", {
        headers: { Authorization: `Bearer ${token}` },
      })
      .then((res) => setFormData(res.data))
      .catch(() =>
        toast({
          title: "Failed to load user data",
          status: "error",
          isClosable: true,
        })
      );
  }, [toast]);

  const handleChange = (e) => {
    const { name, value, type, files } = e.target;
    if (type === "file") {
      setFormData((prev) => ({ ...prev, [name]: files[0] }));
    } else {
      setFormData((prev) => ({ ...prev, [name]: value }));
    }
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    setIsOpen(true);
  };

  const confirmSubmit = () => {
    setIsOpen(false);
    const token = localStorage.getItem("token");
    const form = new FormData();

    Object.entries(formData).forEach(([key, value]) => {
      if (value !== null && value !== undefined) {
        form.append(key, value);
      }
    });

    axios
      .post("http://localhost:5000/profile/me", form, {
        headers: {
          Authorization: `Bearer ${token}`,
          "Content-Type": "multipart/form-data",
        },
      })
      .then(() => {
        toast({
          title: "Profile updated successfully",
          status: "success",
          isClosable: true,
        });
      })
      .catch((error) => {
        console.error(
          "Update error:",
          error.response ? error.response.data : error.message
        );
        toast({
          title: "Update failed",
          description: error.response
            ? error.response.data.message
            : "An unknown error occurred",
          status: "error",
          isClosable: true,
        });
      });
  };

  return (
    // Outer container with gray background
    <Box bg="gray.50" minH="100vh" p={8}>
      {/* Profile Header */}
      <Box
        bg="white"
        p={6}
        rounded="xl"
        shadow="sm"
        display="flex"
        alignItems="center"
        mb={6}
      >
        <Box
          boxSize="80px"
          rounded="full"
          overflow="hidden"
          borderWidth={1}
          mr={4}
        >
          {/* Placeholder or user image */}
          <img
            src={
              formData.profile_picture_url || "https://via.placeholder.com/80"
            }
            alt="Profile"
            style={{ width: "100%", height: "100%", objectFit: "cover" }}
          />
        </Box>

        <Box flex="1">
          <Heading size="md" mb={1}>
            {formData.full_name || "Full Name"}
          </Heading>
          <Box fontSize="sm" color="gray.500">
            {formData.email || "email@example.com"}
          </Box>
        </Box>

        <Button colorScheme="blue" onClick={handleSubmit}>
          Edit
        </Button>
      </Box>

      {/* Form */}
      <Box bg="white" p={6} rounded="xl" shadow="sm">
        <form onSubmit={handleSubmit}>
          <VStack spacing={6} align="stretch">
            {/* Profile Picture */}
            <FormControl>
              <FormLabel>Profile Picture</FormLabel>
              <Input
                type="file"
                name="profile_picture"
                onChange={handleChange}
                accept="image/*"
              />
            </FormControl>

            <Divider />

            {/* Emergency Contact */}
            <SimpleGrid columns={{ base: 1, md: 2 }} spacing={4}>
              <FormControl>
                <FormLabel>Emergency Contact Name</FormLabel>
                <Input
                  name="emergency_contact_name"
                  value={formData.emergency_contact_name}
                  onChange={handleChange}
                />
              </FormControl>
              <FormControl>
                <FormLabel>Emergency Contact Number</FormLabel>
                <Input
                  type="tel"
                  name="emergency_contact_number"
                  value={formData.emergency_contact_number}
                  onChange={handleChange}
                />
              </FormControl>
            </SimpleGrid>

            <Divider />

            {/* Personal Info */}
            <SimpleGrid columns={{ base: 1, md: 2 }} spacing={4}>
              <FormControl isRequired>
                <FormLabel>Full Name</FormLabel>
                <Input
                  name="full_name"
                  value={formData.full_name}
                  onChange={handleChange}
                />
              </FormControl>
              <FormControl>
                <FormLabel>Email</FormLabel>
                <Input
                  type="email"
                  name="email"
                  value={formData.email}
                  onChange={handleChange}
                />
              </FormControl>
              <FormControl>
                <FormLabel>Phone Number</FormLabel>
                <Input
                  name="phone_number"
                  value={formData.phone_number}
                  onChange={handleChange}
                />
              </FormControl>
              <FormControl>
                <FormLabel>Date of Birth</FormLabel>
                <Input
                  type="date"
                  name="date_of_birth"
                  value={formData.date_of_birth}
                  onChange={handleChange}
                />
              </FormControl>
              <FormControl>
                <FormLabel>National ID Number</FormLabel>
                <Input
                  name="national_id_number"
                  value={formData.national_id_number}
                  onChange={handleChange}
                />
              </FormControl>
              <FormControl>
                <FormLabel>Gender</FormLabel>
                <Input
                  as="select"
                  name="gender"
                  value={formData.gender}
                  onChange={handleChange}
                >
                  <option value="">Select</option>
                  <option value="Male">Male</option>
                  <option value="Female">Female</option>
                  <option value="Other">Other</option>
                  <option value="Prefer not to say">
                    Prefer not to say
                  </option>
                </Input>
              </FormControl>
            </SimpleGrid>

            <Divider />

            {/* Address */}
            <Heading size="sm">Address</Heading>
            <SimpleGrid columns={{ base: 1, md: 2 }} spacing={4}>
              <FormControl>
                <FormLabel>Street</FormLabel>
                <Input
                  name="address_street"
                  value={formData.address_street}
                  onChange={handleChange}
                />
              </FormControl>
              <FormControl>
                <FormLabel>City</FormLabel>
                <Input
                  name="address_city"
                  value={formData.address_city}
                  onChange={handleChange}
                />
              </FormControl>
              <FormControl>
                <FormLabel>Province</FormLabel>
                <Input
                  as="select"
                  name="address_province"
                  value={formData.address_province}
                  onChange={handleChange}
                >
                  <option value="">Select</option>
                  <option value="Gauteng">Gauteng</option>
                  <option value="Western Cape">Western Cape</option>
                  <option value="KwaZulu-Natal">KwaZulu-Natal</option>
                  <option value="Eastern Cape">Eastern Cape</option>
                  <option value="Free State">Free State</option>
                  <option value="Limpopo">Limpopo</option>
                  <option value="Mpumalanga">Mpumalanga</option>
                  <option value="North West">North West</option>
                  <option value="Northern Cape">Northern Cape</option>
                </Input>
              </FormControl>
              <FormControl>
                <FormLabel>Postal Code</FormLabel>
                <Input
                  type="number"
                  name="address_postal_code"
                  value={formData.address_postal_code}
                  onChange={handleChange}
                />
              </FormControl>
            </SimpleGrid>

            <Divider />

            {/* Employment & Income */}
            <SimpleGrid columns={{ base: 1, md: 2 }} spacing={4}>
              <FormControl>
                <FormLabel>Employment Status</FormLabel>
                <Input
                  as="select"
                  name="employment_status"
                  value={formData.employment_status}
                  onChange={handleChange}
                >
                  <option value="">Select</option>
                  <option value="Employed">Employed</option>
                  <option value="Unemployed">Unemployed</option>
                  <option value="Student">Student</option>
                  <option value="Retired">Retired</option>
                </Input>
              </FormControl>
              <FormControl>
                <FormLabel>Occupation / Job Title</FormLabel>
                <Input
                  name="occupation"
                  value={formData.occupation}
                  onChange={handleChange}
                />
              </FormControl>
              <FormControl>
                <FormLabel>Monthly Income Range</FormLabel>
                <Input
                  name="monthly_income"
                  value={formData.monthly_income}
                  onChange={handleChange}
                  placeholder="e.g. 10,000 - 20,000"
                />
              </FormControl>
              <FormControl>
                <FormLabel>Employer Name</FormLabel>
                <Input
                  name="employer_name"
                  value={formData.employer_name}
                  onChange={handleChange}
                />
              </FormControl>
            </SimpleGrid>

            <Divider />

            {/* Banking Info */}
            <Heading size="sm">Banking Info</Heading>
            <SimpleGrid columns={{ base: 1, md: 2 }} spacing={4}>
              <FormControl>
                <FormLabel>Bank Name</FormLabel>
                <Input
                  as="select"
                  name="bank_name"
                  value={formData.bank_name}
                  onChange={handleChange}
                >
                  <option value="">Select</option>
                  <option value="Capitec">Capitec</option>
                  <option value="FNB">FNB</option>
                  <option value="ABSA">ABSA</option>
                  <option value="Standard Bank">Standard Bank</option>
                  <option value="Nedbank">Nedbank</option>
                  <option value="Other">Other</option>
                </Input>
              </FormControl>
              <FormControl>
                <FormLabel>Account Number</FormLabel>
                <Input
                  name="bank_account_number"
                  value={formData.bank_account_number}
                  onChange={handleChange}
                />
              </FormControl>
              <FormControl>
                <FormLabel>Account Type</FormLabel>
                <Input
                  as="select"
                  name="bank_account_type"
                  value={formData.bank_account_type}
                  onChange={handleChange}
                >
                  <option value="">Select</option>
                  <option value="Savings">Savings</option>
                  <option value="Cheque">Cheque</option>
                  <option value="Other">Other</option>
                </Input>
              </FormControl>
            </SimpleGrid>

            <Button colorScheme="blue" type="submit" alignSelf="flex-end">
              Save Changes
            </Button>
          </VStack>
        </form>
      </Box>

      <AlertDialog
        isOpen={isOpen}
        leastDestructiveRef={cancelRef}
        onClose={() => setIsOpen(false)}
      >
        <AlertDialogOverlay>
          <AlertDialogContent>
            <AlertDialogHeader fontSize="lg" fontWeight="bold">
              Confirm Changes
            </AlertDialogHeader>
            <AlertDialogBody>
              Are you sure you want to save these changes to your profile?
            </AlertDialogBody>
            <AlertDialogFooter>
              <Button ref={cancelRef} onClick={() => setIsOpen(false)}>
                Cancel
              </Button>
              <Button colorScheme="blue" onClick={confirmSubmit} ml={3}>
                Save
              </Button>
            </AlertDialogFooter>
          </AlertDialogContent>
        </AlertDialogOverlay>
      </AlertDialog>
    </Box>
  );
};

export default PersonalUserPage;

File: gridx_apk/lib/personal_user_page.dart
import 'package:flutter/material.dart';
import 'dart:ui';
import 'package:http/http.dart' as http;
import 'dart:convert';
import 'package:shared_preferences/shared_preferences.dart';

class PersonalUserPage extends StatefulWidget {
  const PersonalUserPage({super.key});

  @override
  State<PersonalUserPage> createState() => _PersonalUserPageState();
}

class _PersonalUserPageState extends State<PersonalUserPage> {
  final TextEditingController _emergencyContactNameController = TextEditingController();
  final TextEditingController _emergencyContactNumberController = TextEditingController();
  final TextEditingController _fullNameController = TextEditingController();
  final TextEditingController _emailController = TextEditingController();
  final TextEditingController _phoneNumberController = TextEditingController();
  final TextEditingController _dateOfBirthController = TextEditingController();
  final TextEditingController _nationalIdNumberController = TextEditingController();
  final TextEditingController _addressStreetController = TextEditingController();
  final TextEditingController _addressCityController = TextEditingController();
  final TextEditingController _addressPostalCodeController = TextEditingController();
  final TextEditingController _occupationController = TextEditingController();
  final TextEditingController _monthlyIncomeController = TextEditingController();
  final TextEditingController _employerNameController = TextEditingController();
  final TextEditingController _bankAccountNumberController = TextEditingController();

  String _gender = '';
  String _addressProvince = '';
  String _employmentStatus = '';
  String _bankName = '';
  String _bankAccountType = '';

  String _profilePictureUrl = 'https://via.placeholder.com/80'; // Placeholder for profile picture

  bool _isLoading = true;
  bool _isSaving = false;
  
  @override
  void initState() {
    super.initState();
    _fetchUserData();
  }

  Future<void> _fetchUserData() async {
    setState(() {
      _isLoading = true;
    });
    final prefs = await SharedPreferences.getInstance();
    final token = prefs.getString('token'); // Assuming token is stored after login

    if (token == null) {
      if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(
          const SnackBar(content: Text('Authentication token not found.')),
        );
      }
      setState(() {
        _isLoading = false;
      });
      return;
    }

    try {
      final response = await http.get(
        Uri.parse('http://10.0.2.2:5000/profile/me'),
        headers: {'Authorization': 'Bearer $token'},
      );

      if (mounted) {
        if (response.statusCode == 200) {
          final Map<String, dynamic> data = jsonDecode(response.body);
          _emergencyContactNameController.text = data['emergency_contact_name'] ?? '';
          _emergencyContactNumberController.text = data['emergency_contact_number'] ?? '';
          _fullNameController.text = data['full_name'] ?? '';
          _emailController.text = data['email'] ?? '';
          _phoneNumberController.text = data['phone_number'] ?? '';
          _dateOfBirthController.text = data['date_of_birth'] ?? '';
          _nationalIdNumberController.text = data['national_id_number'] ?? '';
          _gender = data['gender'] ?? '';
          _addressStreetController.text = data['address_street'] ?? '';
          _addressCityController.text = data['address_city'] ?? '';
          _addressProvince = data['address_province'] ?? '';
          _addressPostalCodeController.text = data['address_postal_code'] ?? '';
          _employmentStatus = data['employment_status'] ?? '';
          _occupationController.text = data['occupation'] ?? '';
          _monthlyIncomeController.text = data['monthly_income'] ?? '';
          _employerNameController.text = data['employer_name'] ?? '';
          _bankName = data['bank_name'] ?? '';
          _bankAccountNumberController.text = data['bank_account_number'] ?? '';
          _bankAccountType = data['bank_account_type'] ?? '';
          _profilePictureUrl = data['profile_picture_url'] ?? 'https://via.placeholder.com/80';
        } else {
          ScaffoldMessenger.of(context).showSnackBar(
            SnackBar(content: Text('Failed to load user data: ${response.statusCode}')),
          );
        }
      }
    } catch (e) {
      if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(
          SnackBar(content: Text('Network error: $e')),
        );
      }
    }
    setState(() {
      _isLoading = false;
    });
  }

  Future<void> _saveUserData() async {
    setState(() {
      _isSaving = true;
    });
    final prefs = await SharedPreferences.getInstance();
    final token = prefs.getString('token');

    if (token == null) {
      if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(
          const SnackBar(content: Text('Authentication token not found.')),
        );
      }
      setState(() {
        _isSaving = false;
      });
      return;
    }

    final Map<String, dynamic> data = {
      'emergency_contact_name': _emergencyContactNameController.text,
      'emergency_contact_number': _emergencyContactNumberController.text,
      'full_name': _fullNameController.text,
      'email': _emailController.text,
      'phone_number': _phoneNumberController.text,
      'date_of_birth': _dateOfBirthController.text,
      'national_id_number': _nationalIdNumberController.text,
      'gender': _gender,
      'address_street': _addressStreetController.text,
      'address_city': _addressCityController.text,
      'address_province': _addressProvince,
      'address_postal_code': _addressPostalCodeController.text,
      'employment_status': _employmentStatus,
      'occupation': _occupationController.text,
      'monthly_income': _monthlyIncomeController.text,
      'employer_name': _employerNameController.text,
      'bank_name': _bankName,
      'bank_account_number': _bankAccountNumberController.text,
      'bank_account_type': _bankAccountType,
      // 'profile_picture': null, // File upload is complex for this step
    };

    try {
      final response = await http.post(
        Uri.parse('http://10.0.2.2:5000/profile/me'),
        headers: {
          'Authorization': 'Bearer $token',
          'Content-Type': 'application/json; charset=UTF-8',
        },
        body: jsonEncode(data),
      );

      if (mounted) {
        if (response.statusCode == 200) {
          ScaffoldMessenger.of(context).showSnackBar(
            const SnackBar(content: Text('Profile updated successfully!')),
          );
        } else {
          ScaffoldMessenger.of(context).showSnackBar(
            SnackBar(content: Text('Failed to update profile: ${response.statusCode}')),
          );
        }
      }
    } catch (e) {
      if (mounted) {
        ScaffoldMessenger.of(context).showSnackBar(
          SnackBar(content: Text('Network error: $e')),
        );
      }
    }

    setState(() {
      _isSaving = false;
    });
  }

  @override
  void dispose() {
    _emergencyContactNameController.dispose();
    _emergencyContactNumberController.dispose();
    _fullNameController.dispose();
    _emailController.dispose();
    _phoneNumberController.dispose();
    _dateOfBirthController.dispose();
    _nationalIdNumberController.dispose();
    _addressStreetController.dispose();
    _addressCityController.dispose();
    _addressPostalCodeController.dispose();
    _occupationController.dispose();
    _monthlyIncomeController.dispose();
    _employerNameController.dispose();
    _bankAccountNumberController.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Stack(
        children: [
          Container(
            decoration: const BoxDecoration(
              image: DecorationImage(
                image: AssetImage('assets/images/AI_suggestion - Copy.png'), // Consistent background
                fit: BoxFit.cover,
              ),
            ),
          ),
          BackdropFilter(
            filter: ImageFilter.blur(sigmaX: 5.0, sigmaY: 5.0),
            child: Container(
              color: const Color.fromARGB(77, 0, 0, 0),
            ),
          ),
          _isLoading
              ? const Center(child: CircularProgressIndicator())
              : SingleChildScrollView(
                  child: Padding(
                    padding: const EdgeInsets.all(24.0),
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: <Widget>[
                        Align(
                          alignment: Alignment.topLeft,
                          child: IconButton(
                            icon: const Icon(Icons.arrow_back, color: Colors.white, size: 30),
                            onPressed: () {
                              Navigator.of(context).pop();
                            },
                          ),
                        ),
                        const SizedBox(height: 40),
                        Align(
                          alignment: Alignment.center,
                          child: Column(
                            children: <Widget>[
                              const Text(
                                "Personal Profile",
                                style: TextStyle(
                                  color: Colors.white,
                                  fontSize: 32,
                                  fontWeight: FontWeight.bold,
                                  fontFamily: 'Montserrat',
                                ),
                                textAlign: TextAlign.center,
                              ),
                              const SizedBox(height: 30),
                              CircleAvatar(
                                radius: 40,
                                backgroundImage: _profilePictureUrl.isNotEmpty
                                    ? NetworkImage(_profilePictureUrl) as ImageProvider<Object>
                                    : const AssetImage('assets/images/avatar_placeholder.png'),
                                backgroundColor: Colors.transparent,
                              ),
                              const SizedBox(height: 10),
                              Text(
                                _fullNameController.text.isEmpty ? "Full Name" : _fullNameController.text,
                                style: const TextStyle(
                                  color: Colors.white,
                                  fontSize: 20,
                                  fontWeight: FontWeight.bold,
                                ),
                              ),
                              Text(
                                _emailController.text.isEmpty ? "email@example.com" : _emailController.text,
                                style: const TextStyle(
                                  color: Colors.white70,
                                  fontSize: 14,
                                ),
                              ),
                              const SizedBox(height: 20),
                            ],
                          ),
                        ),
                        Container(
                          padding: const EdgeInsets.all(20.0),
                          decoration: BoxDecoration(
                            color: Colors.white.withAlpha((0.1 * 255).round()),
                            borderRadius: BorderRadius.circular(15.0),
                            boxShadow: [
                              BoxShadow(
                                color: Colors.black.withAlpha((0.2 * 255).round()),
                                spreadRadius: 1,
                                blurRadius: 3,
                                offset: const Offset(0, 2),
                              ),
                            ],
                          ),
                          child: Column(
                            crossAxisAlignment: CrossAxisAlignment.start,
                            children: <Widget>[
                              // Emergency Contact
                              _buildSectionTitle('Emergency Contact'),
                              _buildTextField(
                                'Emergency Contact Name',
                                _emergencyContactNameController,
                                TextInputType.text,
                              ),
                              const SizedBox(height: 15),
                              _buildTextField(
                                'Emergency Contact Number',
                                _emergencyContactNumberController,
                                TextInputType.phone,
                              ),
                              const SizedBox(height: 25),
                              // Personal Info
                              _buildSectionTitle('Personal Info'),
                              _buildTextField(
                                'Full Name',
                                _fullNameController,
                                TextInputType.text,
                              ),
                              const SizedBox(height: 15),
                              _buildTextField(
                                'Email',
                                _emailController,
                                TextInputType.emailAddress,
                              ),
                              const SizedBox(height: 15),
                              _buildTextField(
                                'Phone Number',
                                _phoneNumberController,
                                TextInputType.phone,
                              ),
                              const SizedBox(height: 15),
                              _buildTextField(
                                'Date of Birth',
                                _dateOfBirthController,
                                TextInputType.datetime,
                              ),
                              const SizedBox(height: 15),
                              _buildTextField(
                                'National ID Number',
                                _nationalIdNumberController,
                                TextInputType.number,
                              ),
                              const SizedBox(height: 15),
                              _buildDropdownField(
                                'Gender',
                                _gender,
                                ['Male', 'Female', 'Other', 'Prefer not to say'],
                                (String? newValue) {
                                  setState(() {
                                    _gender = newValue ?? '';
                                  });
                                },
                              ),
                              const SizedBox(height: 25),
                              // Address
                              _buildSectionTitle('Address'),
                              _buildTextField(
                                'Street',
                                _addressStreetController,
                                TextInputType.streetAddress,
                              ),
                              const SizedBox(height: 15),
                              _buildTextField(
                                'City',
                                _addressCityController,
                                TextInputType.text,
                              ),
                              const SizedBox(height: 15),
                              _buildDropdownField(
                                'Province',
                                _addressProvince,
                                ['Gauteng', 'Western Cape', 'KwaZulu-Natal', 'Eastern Cape', 'Free State', 'Limpopo', 'Mpumalanga', 'North West', 'Northern Cape'],
                                (String? newValue) {
                                  setState(() {
                                    _addressProvince = newValue ?? '';
                                  });
                                },
                              ),
                              const SizedBox(height: 15),
                              _buildTextField(
                                'Postal Code',
                                _addressPostalCodeController,
                                TextInputType.number,
                              ),
                              const SizedBox(height: 25),
                              // Employment & Income
                              _buildSectionTitle('Employment & Income'),
                              _buildDropdownField(
                                'Employment Status',
                                _employmentStatus,
                                ['Employed', 'Unemployed', 'Student', 'Retired'],
                                (String? newValue) {
                                  setState(() {
                                    _employmentStatus = newValue ?? '';
                                  });
                                },
                              ),
                              const SizedBox(height: 15),
                              _buildTextField(
                                'Occupation / Job Title',
                                _occupationController,
                                TextInputType.text,
                              ),
                              const SizedBox(height: 15),
                              _buildTextField(
                                'Monthly Income Range',
                                _monthlyIncomeController,
                                TextInputType.text,
                              ),
                              const SizedBox(height: 15),
                              _buildTextField(
                                'Employer Name',
                                _employerNameController,
                                TextInputType.text,
                              ),
                              const SizedBox(height: 25),
                              // Banking Info
                              _buildSectionTitle('Banking Info'),
                              _buildDropdownField(
                                'Bank Name',
                                _bankName,
                                ['Capitec', 'FNB', 'ABSA', 'Standard Bank', 'Nedbank', 'Other'],
                                (String? newValue) {
                                  setState(() {
                                    _bankName = newValue ?? '';
                                  });
                                },
                              ),
                              const SizedBox(height: 15),
                              _buildTextField(
                                'Account Number',
                                _bankAccountNumberController,
                                TextInputType.number,
                              ),
                              const SizedBox(height: 15),
                              _buildDropdownField(
                                'Account Type',
                                _bankAccountType,
                                ['Savings', 'Cheque', 'Other'],
                                (String? newValue) {
                                  setState(() {
                                    _bankAccountType = newValue ?? '';
                                  });
                                },
                              ),
                              const SizedBox(height: 30),
                              Center(
                                child: ElevatedButton(
                                  onPressed: _isSaving ? null : _saveUserData,
                                  style: ElevatedButton.styleFrom(
                                    backgroundColor: const Color(0xFF80CBC4), // Match sign-in button
                                    shape: RoundedRectangleBorder(
                                      borderRadius: BorderRadius.circular(30.0),
                                    ),
                                    padding: const EdgeInsets.symmetric(horizontal: 40, vertical: 15),
                                  ),
                                  child: _isSaving
                                      ? const CircularProgressIndicator(color: Colors.black)
                                      : const Text(
                                          "Save Changes",
                                          style: TextStyle(
                                            color: Colors.black,
                                            fontSize: 18,
                                            fontWeight: FontWeight.bold,
                                          ),
                                        ),
                                ),
                              ),
                            ],
                          ),
                        ),
                      ],
                    ),
                  ),
                ),
        ],
      ),
    );
  }

  Widget _buildTextField(String label, TextEditingController controller, TextInputType keyboardType) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(
          label,
          style: const TextStyle(
            color: Colors.white70,
            fontSize: 16,
            fontWeight: FontWeight.bold,
          ),
        ),
        const SizedBox(height: 8),
        TextField(
          controller: controller,
          keyboardType: keyboardType,
          style: const TextStyle(color: Colors.white),
          decoration: InputDecoration(
            filled: true,
            fillColor: const Color.fromARGB(25, 255, 255, 255),
            border: OutlineInputBorder(
              borderRadius: BorderRadius.circular(10.0),
              borderSide: BorderSide.none,
            ),
          ),
        ),
      ],
    );
  }

  Widget _buildDropdownField(String label, String? currentValue, List<String> items, Function(String?) onChanged) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(
          label,
          style: const TextStyle(
            color: Colors.white70,
            fontSize: 16,
            fontWeight: FontWeight.bold,
          ),
        ),
        const SizedBox(height: 8),
        Container(
          padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 4),
          decoration: BoxDecoration(
            color: const Color.fromARGB(25, 255, 255, 255),
            borderRadius: BorderRadius.circular(10.0),
          ),
          child: DropdownButtonHideUnderline(
            child: DropdownButton<String>(
              value: currentValue == '' ? null : currentValue,
              isExpanded: true,
              hint: Text(label, style: const TextStyle(color: Colors.white70)),
              icon: const Icon(Icons.arrow_drop_down, color: Colors.white70),
              dropdownColor: const Color(0xFF122849), // Dark background for dropdown
              style: const TextStyle(color: Colors.white, fontSize: 16),
              onChanged: onChanged,
              items: items.map<DropdownMenuItem<String>>((String value) {
                return DropdownMenuItem<String>(
                  value: value,
                  child: Text(value),
                );
              }).toList(),
            ),
          ),
        ),
      ],
    );
  }

  Widget _buildSectionTitle(String title) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Text(
          title,
          style: const TextStyle(
            color: Colors.white,
            fontSize: 20,
            fontWeight: FontWeight.bold,
          ),
        ),
        const SizedBox(height: 15),
      ],
    );
  }
}
